src/librustc/middle/const_eval.rs

```diff
@@ -987,15 +1001,63 @@ pub fn eval_const_expr_partial<'tcx>(tcx: &ty::ctxt<'tcx>,
           } else {
               ty_hint
           };
-          try!(eval_const_expr_partial(tcx, const_expr, item_hint))
+          try!(eval_const_expr_partial(tcx, const_expr, item_hint, fn_args))
       }
+      ast::ExprCall(ref callee, ref args) => {
+          let (
+            decl,
+            unsafety,
+            abi,
+            block,
+          ) = match try!(eval_const_expr_partial(tcx, callee, UncheckedExprNoHint, fn_args)) {
```

Comment:
@eefriedman this `eval_const_expr_partial` gets me the `DefId` of the function I'm calling. I'm not sure if anything but `UncheckedExprNoHint` makes sense here.